CompoundV3-Bots/Reserves-Monitor/src/agent.spec.ts

```diff
@@ -0,0 +1,337 @@
+      JSON.stringify([
+        Finding.from({
+          name: `Comet reserves reached targetReserves`,
+          description: `Reserves on comet contract are >= target reserves`,
+          alertId: "COMP2-1",
+          protocol: "Compound",
+          severity: FindingSeverity.Medium,
+          type: FindingType.Info,
+          metadata: {
+            comet: testConfig.cometAddresses[0],
+            reserves: reservesValue.toString(),
+            targetReserves: reservesValue.toString(),
+          },
+        }),
+      ])
+    );
+  });
+
+  it("returns a finding if Reserves > targetReserves in one comet contract", async () => {
+    let reservesValue: BigNumber = BigNumber.from(500);
+    let targetReservesValue: BigNumber = BigNumber.from(200);
+
+    const blockEvent: BlockEvent = new TestBlockEvent()
+      .setNumber(10)
+      .setTimestamp(INIT_TIMESTAMP);
+
+    for (let i = 1; i < TEST_ADDRESSES.length; i++) {
+      createTargetReservesCall(TEST_ADDRESSES[i], BigNumber.from(100));
+      createGetReservesCall(TEST_ADDRESSES[i], BigNumber.from(50));
+    }
+    createTargetReservesCall(testConfig.cometAddresses[0], targetReservesValue);
+    createGetReservesCall(testConfig.cometAddresses[0], reservesValue);
+
+    const findings = await handleBlock(blockEvent);
+
+    expect(JSON.stringify(findings)).toStrictEqual(
+      JSON.stringify([
+        Finding.fromObject({
+          name: `Comet reserves reached targetReserves`,
+          description: `Reserves on comet contract are >= target reserves`,
+          alertId: "COMP2-1",
+          protocol: "Compound",
+          severity: FindingSeverity.Medium,
+          type: FindingType.Info,
+          metadata: {
+            comet: testConfig.cometAddresses[0],
+            reserves: reservesValue.toString(),
+            targetReserves: targetReservesValue.toString(),
+          },
+        }),
+      ])
+    );
+  });
+
+  it("returns one finding if Reserves > targetReserves and time limit not reached", async () => {
+    let reservesValue: BigNumber = BigNumber.from(500);
+    let targetReservesValue: BigNumber = BigNumber.from(200);
+
+    // First Block
+    const blockEvent: BlockEvent = new TestBlockEvent()
+      .setNumber(10)
+      .setTimestamp(INIT_TIMESTAMP);
+
+    for (let i = 1; i < TEST_ADDRESSES.length; i++) {
+      createTargetReservesCall(TEST_ADDRESSES[i], BigNumber.from(100));
+      createGetReservesCall(TEST_ADDRESSES[i], BigNumber.from(50));
+    }
+
+    createTargetReservesCall(testConfig.cometAddresses[0], targetReservesValue);
+    createGetReservesCall(testConfig.cometAddresses[0], reservesValue);
+
+    // Next Block
+    const blockEvent2: BlockEvent = new TestBlockEvent()
+      .setNumber(11)
+      .setTimestamp(
+        blockEvent.block.timestamp + testConfig.alertFrequency - 10
+      );
+
+    let findings = await handleBlock(blockEvent);
+    findings.concat(await handleBlock(blockEvent2));
```

Comment:
`concat` does not change the array. You probably want `findings.push(...(await handleBlock(blockEvent2)))` or something similar.